### DISCUSSION TITLE
One object in collection is 'Unique'

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#3183] [Questions]
We have a collection object of line Items but one object in the collection is different. The fee/amount associated with that line item needs values from 'Calling code' so we are implementing it similar to what @rockfordlhotka  has discussed on p-129 of Data access book (Using Csla 4 books).
I was curious is it okay to define two methods on collection, 'AddWhateverItem' and 'AddItem'?

Regards

**Comment by rockfordlhotka** [comment]
It is fine, other than that things like datagrid controls will only use `AddItem`.

**Reply by Chicagoan2016** [reply]
@rockfordlhotka if we overload the AddItem with different parameters which one gets called by datagrid control?
Thanks

**Reply by rockfordlhotka** [reply]
iirc, the datagrid controls always call `AddItem()` with no parameters.

It is a missed opportunity by Microsoft, in that they did this by convention instead of by interface (like pretty much everything else in data binding).
